tests/data_context/store/test_datasource_store.py

```diff
@@ -0,0 +1,140 @@
+from pathlib import Path
+
+import pytest
+
+from great_expectations.core.util import convert_to_json_serializable
+from great_expectations.data_context.data_context import DataContext
+from great_expectations.data_context.store.datasource_store import DatasourceStore
+from great_expectations.data_context.types.base import DatasourceConfig
+from great_expectations.data_context.types.resource_identifiers import (
+    ConfigurationIdentifier,
+)
+from great_expectations.util import gen_directory_tree_str
+from tests.test_utils import build_datasource_store_using_filesystem
+
+
+@pytest.fixture
+def datasource_name() -> str:
+    return "my_first_datasource"
+
+
+@pytest.fixture
+def datasource_store_name() -> str:
+    return "datasource_store"
+
+
+@pytest.fixture
+def datasource_key(datasource_name: str) -> ConfigurationIdentifier:
+    return ConfigurationIdentifier(configuration_key=datasource_name)
+
+
+@pytest.fixture
+def empty_datasource_store(datasource_store_name: str) -> DatasourceStore:
+    return DatasourceStore(datasource_store_name)
+
+
+@pytest.fixture
+def datasource_config() -> DatasourceConfig:
+    return DatasourceConfig(
+        class_name="Datasource",
+        execution_engine={"class_name": "PandasExecutionEngine"},
+        data_connectors={
+            "tripdata_monthly_configured": {
+                "class_name": "ConfiguredAssetFilesystemDataConnector",
+                "base_directory": "/path/to/trip_data",
+                "assets": {
+                    "yellow": {
+                        "pattern": r"yellow_tripdata_(\d{4})-(\d{2})\.csv$",
+                        "group_names": ["year", "month"],
+                    }
+                },
+            }
+        },
+    )
+
+
+def test_datasource_store_raises_error_with_invalid_value(
+    empty_datasource_store: DatasourceStore,
+):
+    with pytest.raises(TypeError):
+        empty_datasource_store.set(
+            key="my_first_datasource", value="this is not a datasource"
+        )
+
+
+def test_datasource_store_set_adds_valid_key(
+    empty_datasource_store: DatasourceStore,
+    datasource_config: DatasourceConfig,
+    datasource_key: ConfigurationIdentifier,
+):
+    assert len(empty_datasource_store.list_keys()) == 0
+    empty_datasource_store.set(key=datasource_key, value=datasource_config)
+    assert len(empty_datasource_store.list_keys()) == 1
+
+
+def test_datasource_store_integration(
```

Comment:
It might make sense to mark this as an integration test so we can use flags to exclude/include running it in CI pipelines as we organize our tests a bit more.
```suggestion
@pytest.mark.integration
def test_datasource_store_integration(
```

Reply:
I love that! Great call, I've committed your suggestion.